### PR TITLE
Use new HTTP and Socket API and improve documentation and examples

### PR DESCRIPTION
**Pull request by SimonFrings**
Builds on top of #71, reactphp/socket#263 and reactphp/http#417.

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -67,7 +67,7 @@ public function __construct(LoopInterface $loop = null, $url = null)
             // send everything through a local unix domain socket
             $connector = new FixedUriConnector(
                 $url,
-                new UnixConnector($loop)
+                new UnixConnector()
```
Comment (clue) [inline src/Client.php:70]: Argument should not be omitted if explicit loop is given.

File: src/Client.php [hunk 2]
```diff
@@ -79,7 +79,7 @@ public function __construct(LoopInterface $loop = null, $url = null)
             throw new \InvalidArgumentException('Invalid Docker Engine API URL given');
         }
 
-        $browser = new Browser($loop, $connector);
+        $browser = new Browser($connector);
```
Comment (clue) [inline src/Client.php:82]: Argument should not be omitted if explicit loop is given.

File: tests/FunctionalClientTest.php [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 use Clue\React\Block;
 use Clue\React\Docker\Client;
-use React\EventLoop\Factory as LoopFactory;
+use React\EventLoop\Loop as LoopFactory;
```
Comment (clue) [inline tests/FunctionalClientTest.php:7]: Alias should be updated?

File: tests/IntegrationTest.php [hunk 4]
```diff
@@ -10,12 +10,12 @@ class IntegrationTest extends TestCase
 {
     public function testPingCtorWithExplicitUnixUrlSendsRequestToGivenUnixSocket()
     {
-        $loop = \React\EventLoop\Factory::create();
+        $loop = \React\EventLoop\Loop::get();
```
Comment (clue) [inline tests/IntegrationTest.php:13]: Should we use an import here?

File: tests/IntegrationTest.php [hunk 5]
```diff
@@ -43,17 +43,17 @@ public function testPingCtorWithExplicitUnixUrlSendsRequestToGivenUnixSocket()
 
     public function testPingCtorWithExplicitHttpUrlSendsRequestToGivenHttpUrlWithBase()
     {
-        $loop = \React\EventLoop\Factory::create();
+        $loop = \React\EventLoop\Loop::get();
```
Comment (clue) [inline tests/IntegrationTest.php:46]: Should we use an import here?

**Review by clue — CHANGES_REQUESTED** [review]
@SimonFrings Thanks for looking into this, added some minor remarks below :+1:

File: src/Client.php (hunk 1 above)
Comment (SimonFrings) [reply src/Client.php:70]: :+1:

File: src/Client.php (hunk 2 above)
Comment (SimonFrings) [reply src/Client.php:82]: :+1:

File: tests/IntegrationTest.php (hunk 5 above)
Comment (SimonFrings) [reply tests/IntegrationTest.php:46]: Yes we should!

File: tests/IntegrationTest.php (hunk 4 above)
Comment (SimonFrings) [reply tests/IntegrationTest.php:13]: Yes we should!

File: tests/FunctionalClientTest.php (hunk 3 above)
Comment (SimonFrings) [reply tests/FunctionalClientTest.php]: I removed it, I don't see that it's still needed

**Review by clue — APPROVED** [review]
@SimonFrings Thanks for the update, changes LGTM! :shipit:

**Pull request merged**